File: buffer/gpu/src/jvmMain/java/com/wsr/gpu/JBuffer.java
package com.wsr.gpu;

public class JBuffer {
    public native long allocate(int size, long runtime);
    public native long init(float[] value, long runtime);
    public native void release(long ptr, long runtime);

    public native float[] readAll(long ptr, long runtime);
    public native void write(long ptr, int index, float value, long runtime);
}

File: buffer/gpu/src/jvmAndroidMain/kotlin/com/wsr/knist/gpu/JBuffer.kt
package com.wsr.knist.gpu

object JBuffer {
    external fun allocate(size: Int, runtime: Long): Long
    external fun init(value: FloatArray, runtime: Long): Long
    external fun release(ptr: Long, runtime: Long)

    external fun readAll(ptr: Long, runtime: Long): FloatArray
    external fun write(ptr: Long, index: Int, value: Float, runtime: Long)
}
